lib/ansible/modules/net_tools/lldpad.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, Susant Sahani <susant@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: lldpad
+short_description: Automates lldpad
+description:
+    - Allows you to configure lldpad for various configurations.
+version_added: "2.8"
+options:
+    name:
+        description:
+            - Specifies the name of the interface.
+    command:
+       description: Specifies the command which should be applied.
+       choices: [ "set-lldp", "set-tlv", "enable-tx" ]
+    tlv_identifier:
+        description:
+            - Specifies the type of the TLV which should be configured. Please see 'lldptool --help'
+    tlv_value:
+        description:
+            - Specifies the value what should be applied to the tlv_identifier.
+    admin_status:
+        description:
+            - Specifies the admin staus that should be applied to set-lldp/enable-tx.
+              Note that if command is set-lldp then valid admin_status is one of "tx", "rx", "rxtx" and "disabled" and
+              if command is enable-tx then valid admin_status is one of "yes" or "no".
+        choices: [ "tx", "rx", "rxtx", "disabled", "no", "yes" ]
+
+author: "Susant Sahani (@ssahani) <susant@redhat.com>"
+'''
+
+EXAMPLES = '''
+# Configure LLDP adminStatus to Receive and Transmit for interface eth0
+- lldpad:
+    name=eth0
+    command=set-lldp
+    admin_status=rxtx
+# Set a Management Address TLV on eth0 to carry IPv4 address 192.168.10.10
+- lldpad:
+    name=eth0
+    command=set-tlv
+    tlv_identifier=mngAddr
+    tlv_value=ipv4=192.168.10.10
+# Enable transmit of MTU for interface eth0
+- lldpad:
+    name=eth0
+    command=enable-tx
+    tlv_identifier=MTU
+    admin_status=yes
+# Enable transmit of mngAddr for interface eth0
+- lldpad:
+    name=eth0
+    command=enable-tx
+    tlv_identifier=mngAddr
+    admin_status=yes
+# Enable transmit of macPhyCfg for interface eth0
+- lldpad:
+    name=eth0
+    command=enable-tx
+    tlv_identifier=macPhyCfg
+    admin_status=yes
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import get_platform, AnsibleModule
+
+
+class LldpadModule(object):
```

Comment:
May I suggest that you could inherit `AnsibleModule` if you want to encapsulate all logic in OOP style?

Also, `__metaclass__ = type` allows you do omit explicit inheritance from `object`.

Reply:
Thanks ! Any example where I can look into.

Reply:
These should do:
* https://jfine-python-classes.readthedocs.io/en/latest/metaclass-attribute.html
* https://stackoverflow.com/questions/18554525/what-is-the-purpose-of-metaclass-type

Long story short:
whenever you add
```python
__metaclass__ = type
```
in Python 2, you can replace
```python
class LldpadModule(object):
```
with
```python
class LldpadModule:
```
in that module

Reply:
As for class-based example, you may check out mine:
https://github.com/webknjaz/ansible-gentoo-laptop/blob/master/library/eix.py

Reply:
thanks but do we have any module here and accepted widely ?

Reply:
There's no strict standard about that. Everyone is allowed to have their own mess :)
You asked about example of how it might be done.
I can't remember any module around here of the top of my head and it's kinda hard to analyze 2k modules just to answer you. But I recall someone in our internal slack mentioned that they were doing similar things, so it must be something in this repo if you're up to spending time to search for it.

Anyway, it's just an example of how my ideal module should look like based on principles I want to apply. It doesn't mean that it's the only true way of doing things. But I still think that it's a good architectural example of how one can organize module's control flow and dispatching actions.